### ISSUE TITLE
Document how to implement "reflect a content attribute"

### ISSUE THREAD

**Issue by jdm** [servo/servo#10352]
The macros exist that do this, but there's no clear association between them and the specification text. We should improve our documentation here, and/or rename the macros to make it clearer.

**Comment by perlun** [comment]
Something like this perhaps (it's just a rough start):

> When implementing content attributes for DOM elements, there are a few predefined macros. These macros are implemented here: https://github.com/servo/servo/blob/master/components/script/dom/macros.rs
>
> These macros are especially useful when implementing properties that "must each reflect the respective content attributes of the same name" (quoting the HTML spec). For a DOM attribute described like that in the specification, these macros should _always_ be used.
>
> - `make_getter` - Adds an attribute getter for a `DOMString` attribute.
> - `make_bool_getter` - Adds an attribute getter for a `bool` attribute.
> - `make_limited_int_setter` - **TODO**: document
> - `make_int_getter` - Adds an attribute getter for an `int` attribute.
> - `make_uint_getter` - Adds an attribute getter for an `uint` attribute.
> - etc...

I think it would perhaps be best to keep the documentation for these _in the file_ itself, since there otherwise always a huge risk that we fail in keeping the documentation in sync with the actual implementation.

**Comment by jdm** [comment]
While I see your point, I'm inclined to have text like that present in dom/bindings/mod.rs, which is where related concepts are documented and therefore easier to find and point people at.

**Comment by perlun** [comment]
> While I see your point, I'm inclined to have text like that present in dom/bindings/mod.rs, which is where related concepts are documented and therefore easier to find and point people at.

Fair enough, I don't know the code well enough yet to have strong opinions about it. :smile: We could add a note in `macros.rs` saying that "more info on how to use this can be found in x" then.

**Comment by nox** [comment]
This can probably be salvaged in an `E-Easy` issue.
